src/Context/SoapContext.php

```diff
@@ -273,4 +273,15 @@ public function beforeStepCheckForException(BeforeStepScope $scope)
             throw $this->fault;
         }
     }
+
+    /**
+     * @Given I register the following headers:
+     */
+    public function iRegisterHeaders(TableNode $headers)
+    {
+        foreach ($headers->getRowsHash() as $key => $value) {
+            $soapHeaders[] = new \SoapHeader($key, $key, $value);
```

Comment:
$value is currently a string if taken from TableNode row hash, but it might be an array in real world scenarios, we've used YAML and PyNode for this kind of complex cases in other steps. Would you consider making it recognize arrays? I'm ok if you have a way to put an array inside of table cell as string with JSON and detect json and decode in needed, that can also come as separate feature, but I would expect authentication headers to have "username" and "password" as single header $data param.